From 5fcf430c8105fbeeb07a8cacbae84f941d2a3d55 Mon Sep 17 00:00:00 2001
From: Milan Broz <gmazyland@gmail.com>
Date: Sun, 17 Jun 2018 16:45:09 +0200
Subject: [PATCH] Fix crypto backend for LibreSSL >= 2.7.0.

There are now OpenSSL 1.1.x API functions, no need for compatibility wrapper.

Fixes #393.
---
 lib/crypto_backend/crypto_openssl.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/crypto_backend/crypto_openssl.c b/lib/crypto_backend/crypto_openssl.c
index 4929f82..a6d3667 100644
--- a/lib/crypto_backend/crypto_openssl.c
+++ b/lib/crypto_backend/crypto_openssl.c
@@ -50,9 +50,11 @@ struct crypt_hmac {
 };
 
 /*
- * Compatible wrappers for OpenSSL < 1.1.0
+ * Compatible wrappers for OpenSSL < 1.1.0 and LibreSSL < 2.7.0
  */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || \
+    (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
+
 static void openssl_backend_init(void)
 {
 	OpenSSL_add_all_algorithms();
-- 
2.17.1

